core/GlobalSubstitution.cc

```diff
@@ -118,8 +58,20 @@ GlobalSubstitution::GlobalSubstitution(const GlobalState &from, GlobalState &to,
     to.sanityCheck();
 }
 
-bool GlobalSubstitution::useFastPath() const {
-    return fastPath;
+void GlobalSubstitution::mergeFileTables(const GlobalState &from, GlobalState &to) {
+    UnfreezeFileTable unfreezeFiles(to);
+    int fileIdx = 0; // Skip file 0
+    while (fileIdx + 1 < from.filesUsed()) {
+        fileIdx++;
```

Comment:
this is completely a nit, but to prefer a `while` loop instead of a `for` loop?

```suggestion
    // id 0 is for non-existing FileRef
    for (int fileIdx = 1; fileIdx < from.filesUsed(); fileIdx++) {
```

Reply:
Happy to make this change -- I didn't write this loop :)

My preference would be to import `range-v3` to get `ranges::enumerate` and `ranges::drop`, but that's a bit of a heavyweight change for this PR.